src/Bicep.LangServer/Telemetry/TelemetryProvider.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using OmniSharp.Extensions.LanguageServer.Protocol.Window;
+
+namespace Bicep.LanguageServer.Telemetry
+{
+    public class TelemetryProvider : ITelemetryProvider
+    {
+        private readonly ILanguageServerFacade server;
+
+        public TelemetryProvider(ILanguageServerFacade server)
+        {
+            this.server = server;
+        }
+
+        public void PostEvent(BicepTelemetryEvent telemetryEvent)
+        {
+            if (telemetryEvent is null ||
```

Comment:
telemetryEvent can't be null, neither can telemetryEvent.Properties, so you could simplify this to:
```
if (string.IsNullOrWhiteSpace(telemetryEvent.EventName) || telemetryEvent.Properties.Count == 0)
```

In addition, I feel like it would be better to throw an exception here rather than silently refusing to send the telemetry - that way we're more likely to find potential future code bugs.

Reply:
Sure, updated here:
https://github.com/Azure/bicep/commit/f9547a012329456da53093b7f42b72a2fd9f1044